test/smtchecker.ts

```diff
@@ -0,0 +1,50 @@
+import * as tape from 'tape'
+import smtchecker from '../smtchecker'
+
+tape('SMTChecker', function (t) {
+  t.test('smoke test with no axuiliaryInputRequested', function (st) {
+    var input = {};
+    var output = {};
+    st.equal(smtchecker.handleSMTQueries(input, output), null);
+    st.end();
+  });
+
+  t.test('smoke test with no smtlib2queries', function (st) {
+    var input = {};
+    var output = {auxiliaryInputRequested: {}};
+    st.equal(smtchecker.handleSMTQueries(input, output), null);
+    st.end();
+  });
+
+  t.test('smoke test with empty smtlib2queries', function (st) {
+    var input = {};
+    var output = {auxiliaryInputRequested: {smtlib2queries: {}}};
+    st.equal(smtchecker.handleSMTQueries(input, output), null);
+    st.end();
+  });
+
+  t.test('smtCallback should return type function', (st) => {
+    const response = smtchecker.smtCallback(() => {});
+    st.equal(typeof response, 'function')
+    st.end()
+  })
+
+  t.test('smtCallback should error when passed parser fails', (st) => {
```

Comment:
These aren't part of `smtchecker.js`?

Reply:
what do you mean?

these are testing this implementation

```
export function smtCallback (solver) {
  return function (query) {
    try {
      const result = solver(query);
      return { contents: result };
    } catch (err) {
      return { error: err };
    }
  };
}
```

and is apart of the smtchecker. Test coverage had to be increased since now its based on the compiled output and was failing to meet the min requirements.

Reply:
I mean that this test doesn't show up in `smtchecker.js`. Since this is coverage improvement, it should not be hidden in a big refactoring PR.

Reply:
This PR will not go through without an introduction of a test case.

The tests have to be ran against the transpiled code since there is a performance problem with Typescript and dynamic imports. The transpiled code is less than the original code and thus with the same exists passing results in a lower test coverage and blocks the PR from being mergeable.

Reply:
This could at least be a separate commit.

Reply:
This was merged as part of #565.